components/style/matching.rs

```diff
@@ -445,7 +445,10 @@ trait PrivateMatchMethods: TElement {
         let mut animation_state = animation_states.remove(&this_opaque).unwrap_or_default();
 
         if let Some(ref mut old_values) = *old_values {
-            animation_state.compute_before_change_style::<Self>(
+            // We convert old values into `before-change-style` here.
+            let old_values = Arc::make_mut(old_values);
```

Comment:
So, this will copy the style almost surely, right? It'd be nice to not do that if there are no animations running.

Reply:
Oh, nice catch! I've modified this so that the methods now take an `Arc` which will ensure that callers don't accidentally clone the style like I did in this version of the change.